src/data/Data.js

```diff
@@ -680,17 +695,20 @@ export const WorkspaceData = _.flow(
 
     const loadWdsData = useCallback(async () => {
       // Try to load the proxy URL
-      if (!wdsProxyUrl || !['Ready', 'Error'].includes(wdsProxyUrl.status)) {
-        const wdsUrl = await loadWdsUrl(workspaceId);
-        if (wdsUrl) {
-          await loadWdsTypes(wdsUrl, workspaceId);
+      if (!wdsApp || !['Ready', 'Error'].includes(wdsApp.status)) {
+        const wdsApp = await loadWdsApp(workspaceId);
+        // TODO: figure out how not to make this redundant fetch, per
```

Comment:
I struggled with trying to address this feedback from the prior PR - I think because I am not fully grok the react lifecycle that makes the current ugly extra fetch necessary.